buildSrc/src/main/kotlin/code-generation.gradle.kts

```diff
@@ -97,20 +97,28 @@ val createE2eTasksForProvider by extra {
         val version = schema.kotlinVersion
         val javaLibraryDependency = "com.pulumi:$providerName:${schema.javaVersion}"
 
-        val sourceSetName = "pulumi${providerName.capitalized()}E2e"
+        val sourceSetName = "pulumi${schema.versionedProviderName.capitalized()}E2e"
         val generationTaskName = "generate${sourceSetName.capitalized()}Sources"
         val compilationTaskName = "compile${sourceSetName.capitalized()}Kotlin"
         val jarTaskName = "${sourceSetName}Jar"
         val implementationDependency = "${sourceSetName}Implementation"
         val archiveName = "pulumi-$providerName-kotlin"
-        val downloadTaskName = "download${providerName.capitalized()}E2eSchema"
+        val downloadTaskName = "download${schema.versionedProviderName.capitalized()}E2eSchema"
 
-        val schemaDownloadPath =
-            Paths.get(rootDir, "build", "tmp", "schema", "e2e", "$providerName-$version.json").toFile()
+        val schemaDownloadPath = Paths.get(
+            rootDir, "build", "tmp", "schema", "e2e", "${schema.versionedProviderName}-$version.json",
```

Comment:
(probably minor)

We already have the version `...-$version.json`. Maybe we should use a normal provider name then (`${schema.providerName}-$version.json`).

Reply:
True, we don't need that here